gensim/similarities/docsim.py

```diff
@@ -539,7 +539,7 @@ def convert(shard_no, doc):
             if not is_corpus:
                 # user asked for num_best most similar and query is a single doc
                 results = (convert(shard_no, result) for shard_no, result in enumerate(shard_results))
-                result = heapq.nlargest(self.num_best, itertools.chain(*results), key=lambda item: item[1])
+                result = heapq.nlargest(self.num_best, itertools.chain(*results), key=lambda item: abs(item[1]))
```

Comment:
This is getting a little bit too hairy. We're multiple indentation levels in, and we're duplicating a bit of code. Let's split out a function:

```python
def _nlargest(n, results):
    return heapq.nlargest(n, itertools.chain(*results), key=lambda item: abs(item[1]))
```

and then use it inside as:

```suggestion
                result = _nlargest(self.num_best, results)
```

Reply:
Can `lambda` accept a tuple argument? If so, `key=lambda (feature, weight): abs(weight)` more readable.

Reply:
@piskvorky Unfortunately it was removed from python3 (you can check it [here](https://www.python.org/dev/peps/pep-3113/))

@mpenkov I guess `_nlargest()` can be tested, whereas creating test for the whole `Similarity` object is complicated (I guess it involves changing the corpus, that is used for testing and fixing all other test OR maybe adding explicit test and corpus for `Similarity` that checks, that right objects are extracted with `_nlargest`). Which is more preferable?

Reply:
Write tests for nlargest for now.